resolver/resolver.cc

```diff
@@ -760,7 +760,18 @@ class ResolveConstantsWalk {
         if (send.recv.isSelfReference() && send.fun == core::Names::mixesInClassMethods()) {
             auto item = ClassMethodsResolutionItem{ctx.file, ctx.owner, &send};
             this->todoClassMethods_.emplace_back(move(item));
+        } else {
+            auto recvAsConstantLit = ast::cast_tree<ast::ConstantLit>(send.recv);
+            if (recvAsConstantLit != nullptr) {
+                auto original = ast::cast_tree<ast::UnresolvedConstantLit>(recvAsConstantLit->original);
+                if (original != nullptr && original->cnst == core::Names::Constants::Magic() &&
+                    send.fun == core::Names::mixesInClassMethods()) {
+                    auto item = ClassMethodsResolutionItem{ctx.file, ctx.owner, &send};
+                    this->todoClassMethods_.emplace_back(move(item));
+                }
+            }
```

Comment:
I couldn't find a way to check for the constant name `Magic` without the double casting, as a result it got messy here. Open to suggestions.

Reply:
See my suggested changes.

Feel free to play around with them, in case you want to further simplify anything.